scripts/docker_deploy_manual.sh

```diff
@@ -2,26 +2,43 @@
 
 set -e
 
-if [ -z "$1" ]
-then
-    echo "Usage: $0 <tag/branch>"
-    exit 1
-fi
+REPO_ROOT="$(dirname "$0")/.."
+# shellcheck source=scripts/common.sh
+source "${REPO_ROOT}/scripts/common.sh"
+
 image="ethereum/solc"
-branch="$1"
+
+if (( $# == 0 || $# >= 4 )); then
+    fail "Usage: $0 <tag/branch> [repo URL] [--no-push]"
+else
+    branch="$1"
+fi
+
+if (( $# >= 2 )); then
+    repo_url="$2"
+else
+    repo_url=https://github.com/ethereum/solidity.git
+fi
+
+if (( $# >= 3 )); then
+    [[ $3 == --no-push ]] || fail "Invalid flag: $3. Expected --no-push."
+    publish=false
+else
+    publish=true
+fi
```

Comment:
I guess we could rewrite this part to something like:
```suggestion

if (( $# < 1 || $# > 3 )); then
    fail "Usage: $0 <tag/branch> [repo URL] [--no-push]"
fi

branch="$1"
repo_url="${2:-https://github.com/ethereum/solidity.git}"
if (( $# == 3 )); then
    [[ $3 == --no-push ]] || fail "Invalid flag: $3. Expected --no-push."
    publish=false
else
    publish=true
fi
```

Reply:
True, that's better.

Done.